packages/react-query/src/createTRPCReact.tsx

```diff
@@ -54,31 +87,41 @@ export interface ProcedureUseQuery<
 > {
   <
     TQueryFnData extends inferTransformedProcedureOutput<TProcedure> = inferTransformedProcedureOutput<TProcedure>,
-    TData extends inferTransformedProcedureOutput<TProcedure> = inferTransformedProcedureOutput<TProcedure>,
+    TData = TQueryFnData,
+    TQueryData extends inferTransformedProcedureOutput<TProcedure> = TQueryFnData,
```

Comment:
This exposes a 3rd generic that people probably shouldn't use. We might want to hide it (which I think is what RQ does with `UseQueryOptions` wrapping `UseBaseQueryOptions`). Or we can just name it `__BadIdeaDontUseThisType__`